### PR TITLE
Encd 3504 genetic mods ui

### PR DESCRIPTION
**Pull request by estherc**
This is a whopper of a PR, combining all the schema changes, upgrades, tests, changelog updates, manually migrated data and Forrest's corresponding UI changes related to GMs. GMs, constructs, RNAis, TALEs and CRISPR objects should have one or more required properties made import_only so we do not get any new submissions for the rollout. The GM upgrade test is commented out because it could not be made to work. GM dependencies are described in the comments. URLs were made into a mixin and mixed into donors. Upgrade only covers existing GM objects that are linked to existing CRISPR and TALE objects, all other migrations (construct, RNAi) have to be done manually.

### CODE REVIEW

File: src/encoded/schemas/genetic_modification.json [hunk 1]
```diff
@@ -488,7 +488,6 @@
                 "properties": {
                     "name": {
                         "type": "string",
-                        "default": "",
```
Comment (hitz) [inline src/encoded/schemas/genetic_modification.json]: name is required with a default?  of ""?  That seems not super useful.

File: src/encoded/types/biosample.py [hunk 2]
```diff
@@ -306,20 +306,20 @@ def model_organism_donor_modifications(self, request, donor=None):
         "description": "All genetic modifications made in either the donor and/or biosample.",
         "type": "array",
         "items": {
-            "title": "Applied modification",
+            "title": "All applied modification",
```
Comment (hitz) [inline src/encoded/types/biosample.py]: grammar on this is a little odd?

File: src/encoded/types/biosample.py [hunk 3]
```diff
@@ -306,20 +306,20 @@ def model_organism_donor_modifications(self, request, donor=None):
...
             "description": "Genetic modification made in either the donor and/or biosample.",
             "coment": "See genetic_modification.json for available identifiers.",
             "type": "string",
             "linkTo": "GeneticModification",
         }
-    }, define=True)
```
Comment (hitz) [inline src/encoded/types/biosample.py]: what does define=True do here?

**Comment by estherc** [comment]
Those issues are all fixed in the latest commits.

**Pull request merged**